Clarify usage of `_expected` querystring parameter (#600)
mozilla/remote-settings: docs/client-specifications.rst

.. note::

>     The ``_expected={}`` querystring parameter is mandatory. Either you pass the current collection timestamp value obtained when polling for changes in order to bust the CDN cache, or you use a hard-coded value (eg. ``0``) and rely on the cache TTL. See section below about cache busting.

Reviewer: ```suggestion
    The ``_expected={}`` querystring parameter is mandatory. Either you pass the timestamp value obtained from the last fetch for the collection in order to bust the CDN cache, or you use a hard-coded value (eg. ``0``) and rely on the cache TTL. See section below about cache busting.
```

>     The ``_expected={}`` querystring parameter is mandatory. Either you pass the current collection timestamp value obtained when polling for changes in order to bust the CDN cache, or you use a hard-coded value (eg. ``0``) and rely on the cache TTL. See section below about cache busting.

Reviewer: I'm not quite sure what "current collection timestamp" means.  Is it the timestamp of the last fetch?  The timestamp returned in the data for the last fetch?  The current time?  Whichever one it is, please spell out how it would work to bust the cache.

Can you lay out a scenario when you would want to bust the cache while polling?  In general, it seems like I wouldn't want to do that.

Also, how can we use the `_since` param in a cache-friendly manner?  If all clients did something naive like use the timestamp they last polled, then it seems like each request would be a cache-miss.

.. note::

>     The ``_expected={}`` querystring parameter is mandatory. Either you receive a Push notification from the server, and pass the timestamp value in order to bust the CDN cache, or you use a hard-coded value (eg. ``0``) and rely on the cache TTL. See section below about cache busting.

Reviewer: This one makes total sense to me.  Is this param is intended more for responding to push notifications than for polling?
